http-api-monitor-offers.adoc

```diff
@@ -0,0 +1,327 @@
+
+=== Get the market price
+
+In order to get the market price of BTC in EUR, execute the following query:
+
+    curl http://localhost:8080/api/v1/currencies/prices?currencyCodes=EUR
+
+You should receive a response like the following:
+
+[source,json]
+----
+{
+  "prices": {
+    "EUR": 7035.62
+  }
+}
+----
+
+== The JavaScript part
+
+Let's install some dependencies:
+
+    npm install lodash http-as-promised nodemailer
+
+In general our script will look like this:
+
+[source,javascript]
+----
+Promise.all([getOffers(), getMarketPrice()])
+        .then(filterOffers)
+        .then(notify)
+        .catch(e => console.error(e));
+----
+
+So first 2 things to do (concurrently and asynchronously) is to fetch offers and market price of BTC from our API. Then we need to filter those offers and finally send notification.
+
+Getting offers is a simple call that returns promise with array of offers:
+
+[source,javascript]
+----
+function getOffers() {
+    return $http.get('http://localhost:8080/api/v1/offers', {resolve: 'body', json: true})
+        .then(body => body.offers);
+}
+----
+
+Similarly with `getMarketPrice`:
+
+[source,javascript]
+----
+function getMarketPrice() {
+    return $http.get('http://localhost:8080/api/v1/currencies/prices?currencyCodes=EUR', {resolve: 'body', json: true})
+            .then(body => _.get(body, 'prices.EUR'))
+}
+----
+
+Now our `filterOffers` function is ready to receive an array of results from the two functions described above:
+
+[source,javascript]
+----
+function filterOffers([offers, marketPrice]) {
+    return _(offers)
+            .filter({
+                baseCurrencyCode: 'BTC',
+                counterCurrencyCode: 'EUR',
+                direction: 'SELL',
+                paymentMethodId: 'SEPA'
+            })
+            .filter(i => _.includes(i.acceptedCountryCodes, 'PL'))
+            .filter(getPriceFilter(marketPrice))
+            .map(i => ({amount: i.amount, margin: i.useMarketBasedPrice ? i.marketPriceMargin : marketPrice / i.price}))
+            .value();
+}
+----
+
+This function filters offers to match our criteria. It returns matching offers and maps them to a bit simpler structure that contains as little data as needed for `notify` function. We are using `lodash` library to simplify the filtering.
+
+The `getPriceFilter` function creates the actual filter function and looks like this:
+
+[source,javascript]
+----
+function getPriceFilter(marketPrice) {
+    const maxPrice = marketPrice * (1 - threshold) * 10000;
+    return offer => {
+        if (offer.useMarketBasedPrice)
+            return offer.marketPriceMargin >= threshold;
+        return offer.price < maxPrice;
+    }
+}
+----
+
+We are multiplying `marketPrice` by `10000` because that is the format in which the API returns the price.
```

Comment:
- [ ] Somewhere right about here, you should instruct the user to actually run `node bisqmon.js` (or whatever the correct invocation should be), and then show them what they should expect to see as correct output. This is the big payoff. They've hung with us for 15 minutes, reading our instructions, taking in the context, and following each step faithfully. Now they get what they came for: working code that does exactly what we said it would.

Reply:
done